### PR TITLE
Eliminte the useless row collect.

### PR DESCRIPTION
**Pull request by Shylock-Hg**
#### What type of PR is this?
- [ ] bug
- [ ] feature
- [x] enhancement

#### Checklist：
- [ ] Documentation affected （Please add the label if documentation needs to be modified.)
- [ ] Incompatible （If it is incompatible, please describe it and add corresponding label.）
- [ ] Need to cherry-pick （If need to cherry-pick to some branches, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

#### Release notes：
Please confirm whether to reflect in release notes and how to describe:
>                                                                 `

### CODE REVIEW

**Review by yixinglu — COMMENTED** [review]
Maybe we could also revise the DataCollect Node generation rule, check whether this node is not necessary in some scenarios. WDYT?

**Comment by Shylock-Hg** [comment]
> 

The plan generated by planner is `DataCollect -> Limit -> Project`, after optimization it's `DataCollect -> Project -> Limit`. So in planner, we don't know it's useless.

**Comment by Shylock-Hg** [comment]
> Maybe we could also revise the DataCollect Node generation rule, check whether this node is not necessary in some scenarios. WDYT?

The plan generated by planner is `DataCollect -> Limit -> Project`, after optimization it's `DataCollect -> Project -> Limit`. So in planner, we don't know it's useless.

**Comment by yixinglu** [comment]
> > Maybe we could also revise the DataCollect Node generation rule, check whether this node is not necessary in some scenarios. WDYT?
> 
> The plan generated by planner is `DataCollect -> Limit -> Project`, after optimization it's `DataCollect -> Project -> Limit`. So in planner, we don't know it's useless.

I mean you can check when we need to append the DataCollect plan node into the plan, maybe now it's not necessary to do that.

**Review by CPWstatic — APPROVED** [review]

**Review by jievince — APPROVED** [review]

**Comment by czpmango** [comment]
> Maybe we could also revise the DataCollect Node generation rule, check whether this node is not necessary in some scenarios. WDYT?

IMHO, it's unnecessary in most cases, bcz `Project` may also works. Besides, introducing redundant plannode types is not optimizer friendly.
https://github.com/vesoft-inc/nebula/blob/32394f4b767c66bfd067fcfe6f928988fd58a61f/src/graph/planner/SequentialPlanner.cpp#L39-L60

File: src/graph/optimizer/rule/EliminateRowCollectRule.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+// TODO match DataCollect->(Any Node with real result)
+const Pattern& EliminateRowCollectRule::pattern() const {
+  static Pattern pattern = Pattern::create(graph::PlanNode::Kind::kDataCollect,
+                                           {Pattern::create(graph::PlanNode::Kind::kProject)});
+  return pattern;
+}
+
+bool EliminateRowCollectRule::match(OptContext* octx, const MatchedResult& matched) const {
```
Comment (czpmango) [inline src/graph/optimizer/rule/EliminateRowCollectRule.cpp:35]: Maybe `EliminateNoopDataCollect`?

**Pull request merged**